python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -1,30 +1,173 @@
+        # because AIRFLOW_HOME has been overriden airflow needs to be reloaded
+        if is_airflow_2_loaded_in_environment():
+            importlib.reload(airflow.configuration)
+            importlib.reload(airflow.settings)
+            importlib.reload(airflow)
+        else:
+            importlib.reload(airflow)
+        if not airflow_initialized:
+            db.initdb()
+            create_airflow_connections(
+                [Connection(**c) for c in context.resource_config["connections"]]
+            )
+
+
+def _create_airflow_db_resource(connections: Optional[List[Connection]] = None):
+    serialized_connections = serialize_connections(connections)
+    airflow_db_resource_def = ResourceDefinition(
+        resource_fn=airflow_db,
+        required_resource_keys={"dag"},
+        config_schema={
+            "connections": Field(
+                Array(inner_type=dict),
+                default_value=serialized_connections,
+                is_required=False,
+            ),
+        },
+        description="Airflow DB used by wrapped airflow tasks",
+    )
+    return airflow_db_resource_def
+
+
+def get_dagrun(context: InitResourceContext) -> DagRun:
+    airflow_home_path = os.path.join(tempfile.gettempdir(), f"dagster_airflow_{context.run_id}")
+    os.makedirs(airflow_home_path, exist_ok=True)
+    with Locker(airflow_home_path):
+        dag = context.resources.dag
+        run_tags = context.dagster_run.tags if context.dagster_run else {}
+        if AIRFLOW_EXECUTION_DATE_STR in run_tags:
+            execution_date = _parse_execution_date_for_job(dag, run_tags)
+        elif "dagster/partition" in run_tags:
+            execution_date = _parse_execution_date_for_asset(dag, run_tags)
+        else:
+            raise DagsterInvariantViolationError(
+                'Could not find "{AIRFLOW_EXECUTION_DATE_STR}" in tags "{tags}". Please '
+                'add "{AIRFLOW_EXECUTION_DATE_STR}" to tags before executing'.format(
+                    AIRFLOW_EXECUTION_DATE_STR=AIRFLOW_EXECUTION_DATE_STR,
+                    tags=run_tags,
+                )
+            )
+        dagrun = dag.get_dagrun(execution_date=execution_date)
+        if not dagrun:
+            if is_airflow_2_loaded_in_environment():
+                dagrun = dag.create_dagrun(
+                    state=DagRunState.RUNNING,
+                    execution_date=execution_date,
+                    run_type=DagRunType.MANUAL,
+                )
+            else:
+                dagrun = dag.create_dagrun(
+                    run_id=f"dagster_airflow_run_{execution_date}",
+                    state=State.RUNNING,
+                    execution_date=execution_date,
+                )
+        return dagrun
+
+
+def _create_dagrun_resource() -> ResourceDefinition:
+    airflow_db_resource_def = ResourceDefinition(
+        resource_fn=get_dagrun,
+        required_resource_keys={"dag", "airflow_db"},
```

Comment:
I still don't see these accesses in the initialization function.

Is the sole purpose to ensure the init functions run in a certain order?